### ISSUE TITLE
[Bug report] note.content = "value" does not update note

### ISSUE THREAD

**Issue by jwhonce** [mm21/trilium-alchemy#5]
**Describe the bug**
For an existing note assigning a new value to content does not update the note in Trilium.

**To Reproduce**
```python
        trilium.flush()
        print("Updating Content Dirty Set", trilium.dirty_set)
        task.content = content_template.substitute(
            {
                "key": ticket.key,
                "summary": ticket.summary,
                "priority": ticket.priority,
                "status": ticket.status,
                "created": ticket.created.strftime("%Y-%m-%d"),
                "url": ticket.url,
                "updated": ticket.updated.strftime("%Y-%m-%d"),
            }
        )
        print("Content Dirty Set", trilium.dirty_set)
        trilium.flush()
```
Logging output:
```
Updating Content Dirty Set set()
Content Dirty Set set()
INFO:root:Flushing 0 entities: (create/update/delete) 0/0/0 notes, 0/0/0 attributes, 0/0/0 branches
```

**Expected behavior**
I would expect setting note.content to cause the new content to update given trilium note.

**Screenshots**

**Environment**
 - OS: [e.g. Ubuntu] Fedora
 - Python version 3.11.4
 - TriliumAlchemy version 0.1.3

**Additional context**
1. Using Note(content="value") works as expected.
2. Adding a `task["attribute"] = "value"` is logged and flushed, but `content` is still not changed in trilium.

**Comment by mm21** [comment]
Hi, thanks for reporting this. Strange since there are test cases setting `Note.content`.

If you don't mind, the following info will help:

- Is it a newly created note, or one which already exists? I just realized the content tests are using only pre-existing notes, not newly created ones.
- If it already exists, is there any chance the content is unchanged? If the content is unchanged it will skip setting the note as dirty.

**Comment by jwhonce** [comment]
Let me test both cases. I will remove `Note(content=...)` and just use the setter.

**Comment by jwhonce** [comment]
I have added `flush()` calls to aid when reading the logging output. There are three candidates, two existing (created by this code) and one new one (I keep deleting to test). The content was not updated on any of them.

```python
candidate = trilium.search(
            f'#task #!doneDate #jiraKey="{ticket.key}"', ancestor_note=todo_root
        )
        match len(candidate):
            case 0:
                print(f"New: {ticket.key}")

                title = ticket.summary[:76] + "..." * (len(ticket.summary) > 76)
                task = Note(
                    title=f"{ticket.key}: {title}",
                    template=todo_template
                )
                task["jiraKey"] = ticket.key
                task["todoDate"] = datetime.now().strftime("%Y-%m-%d")
                task["location"] = "work"
                task["updated"] = ticket.updated.strftime("%Y-%m-%d")
                task += [Label("tag", "jira")]

                task["jiraPriority"] = ticket.priority
  
                todo_root += task
                trilium.flush()

                task ^= (today, "TODO")
            case 1:
                print(f"Update: {ticket.key}")
                task = candidate[0]
                if ticket.resolved:
                    task["doneDate"] = ticket.resolved.strftime("%Y-%m-%d")

            case _:
                typer.echo(f"Multiple matches for {ticket.key}", err=True)
                raise typer.Abort()

        task["jiraPriority"] = ticket.priority
        task["jiraStatus"] = ticket.status

        trilium.flush()

        print("Updating Content Dirty Set", trilium.dirty_set)
        task.content = content_template.substitute(
            {
                "key": ticket.key,
                "summary": ticket.summary,
                "priority": ticket.priority,
                "status": ticket.status,
                "created": ticket.created.strftime("%Y-%m-%d"),
                "url": ticket.url,
                "updated": ticket.updated.strftime("%Y-%m-%d"),
            }
        )
        print("Content Dirty Set", trilium.dirty_set)
        task.flush()
```

**Comment by mm21** [comment]
Think I found the problem - you may be using an older version of Trilium which doesn't provide `blobId` in ETAPI. This is used to check if the note content changed, but if it isn't provided TriliumAlchemy falls back to comparing the note content itself, which has a bugged comparison. Since I haven't tested on a version of Trilium without `blobId` it never came up.

Added a debug log to confirm if this is the case. Please do reopen if it doesn't fix the issue.

**Comment by mm21** [comment]
Released in `trilium_alchemy-0.1.4`.
